From e5bb95ca605cf4a3abe1926954c4952a22a2ee7e Mon Sep 17 00:00:00 2001
From: michael-west <michael.west@ettus.com>
Date: Fri, 16 May 2014 15:50:14 -0700
Subject: [PATCH 24/31] Fix for BUG 456:  LED TX/RX colors backwards on X300 -
 Corrected bit masks so TX will light the red LED and
 RX will light the green LED

---
 host/lib/usrp/x300/x300_impl.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/host/lib/usrp/x300/x300_impl.cpp b/host/lib/usrp/x300/x300_impl.cpp
index e931b79..698d98f 100644
--- a/host/lib/usrp/x300/x300_impl.cpp
+++ b/host/lib/usrp/x300/x300_impl.cpp
@@ -1321,8 +1321,8 @@ void x300_impl::update_atr_leds(gpio_core_200_32wo::sptr leds, const std::string
 {
     const bool is_txrx = (rx_ant == "TX/RX");
     const int rx_led = (1 << 2);
-    const int txrx_led = (1 << 1);
-    const int tx_led = (1 << 0);
+    const int tx_led = (1 << 1);
+    const int txrx_led = (1 << 0);
     leds->set_atr_reg(dboard_iface::ATR_REG_IDLE, 0);
     leds->set_atr_reg(dboard_iface::ATR_REG_RX_ONLY, is_txrx? txrx_led : rx_led);
     leds->set_atr_reg(dboard_iface::ATR_REG_TX_ONLY, tx_led);
-- 
1.7.10.4

